Program.cs

```diff
@@ -24,8 +24,44 @@ class MyWindow : Window {
       mStride = mBmp.BackBufferStride;
       image.Source = mBmp;
       Content = image;
+      image.MouseDown += OnMouseDown;
+   }
+
+   void OnMouseDown (object sender, MouseButtonEventArgs e) {
+      var pos = e.GetPosition (this);
+      mStart = mEnd == Zero ? pos : mEnd;
+      mEnd = pos;
+      DrawLine ((int)mStart.X, (int)mStart.Y, (int)mEnd.X, (int)mEnd.Y);
+   }
+   Point mStart = new (), mEnd = Zero;
+   static Point Zero = new (0, 0);
 
-      DrawMandelbrot (-0.5, 0, 1);
+   void DrawLine (int x1, int y1, int x2, int y2) {
+      try {
+         mBmp.Lock ();
+         mBase = mBmp.BackBuffer;
+         int dx = Math.Abs (x2 - x1), dy = -Math.Abs (y2 - y1);

```

Comment:
Why reverse the sign of dy? That is an implementation detail that is handled in SetPixel - the rest of the code should not care that Y in the bitmap goes 'downwards'. Because of this, the code is not symmetrical for dx and dy.